astroid/brain/brain_pathlib.py

```diff
@@ -0,0 +1,51 @@
+# Licensed under the LGPL: https://www.gnu.org/licenses/old-licenses/lgpl-2.1.en.html
+# For details: https://github.com/PyCQA/astroid/blob/main/LICENSE
+# Copyright (c) https://github.com/PyCQA/astroid/blob/main/CONTRIBUTORS.txt
+
+from __future__ import annotations
+
+from collections.abc import Iterator
+
+from astroid import bases, context, extract_node, inference_tip, nodes
+from astroid.exceptions import InferenceError, UseInferenceDefault
+from astroid.manager import AstroidManager
+
+PATH = extract_node(
+    """from pathlib import Path
+Path
+"""
+)
+PATH_CLASSDEF: nodes.ClassDef = next(PATH.infer())
+
+
+def _looks_like_parents_subscript(node: nodes.Subscript) -> bool:
+    if not isinstance(node.value, nodes.Name) and not (
+        isinstance(node.value, nodes.Attribute) and node.value.attrname == "parents"
+    ):
```

Comment:
```suggestion
    if not (
        isinstance(node.value, nodes.Name)
        or isinstance(node.value, nodes.Attribute)
        and node.value.attrname == "parents"
    ):
```
`not ... and not (... and ...)` is quite complicated and difficult to follow.